d1lod/d1lod/sesame/interface.py

```diff
@@ -573,12 +577,15 @@ def deleteDatasetTriples(self, identifier):
         Find the blank node for the identifier of this dataset and delete
         all statements about it.
         """
-        query = u"""DELETE
-        WHERE {
-        <%s> <%s> ?identifier .
-        ?identifier ?s ?p
+        query = u"""DELETE WHERE
+        {
+            GRAPH <%s>
+            {
+                <%s> <%s> ?identifier .
+                ?identifier ?s ?p
+            }
         }
-        """ % (dataset, has_identifier)
+        """ % (self.repository.name, dataset, has_identifier)
```

Comment:
I see lots of explicitly passing in the repository (graph) name to functions handling the execution of SPARQL queries. Since you're passing `self.repository_name` to another method on `self`, which already knows which `repository` it's pointing at, whichever method you're calling could just automatically know which repository you mean. Do these method signatures need the flexibility to insert triples into another graph?

Calls like `self.repository.insert_data(repository_name=self.repository.name` seem like we're missing an opportunity to be more concise like `self.repository.insert_data(dataset, something)`

Reply:
Removed functionality to be able to insert another graph into the DB.
Commit: https://github.com/ec-geolink/d1lod/pull/48/commits/a6e0e1f40f2d88ab3f8211431534a8bf1174aaee